integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonPreAggregateRules.scala

```diff
@@ -1264,13 +1264,16 @@ object CarbonPreAggregateDataLoadingRules extends Rule[LogicalPlan] {
   override def apply(plan: LogicalPlan): LogicalPlan = {
     val validExpressionsMap = scala.collection.mutable.LinkedHashMap.empty[String, NamedExpression]
     plan transform {
-      case aggregate@Aggregate(_, aExp, _) if validateAggregateExpressions(aExp) =>
+      case aggregate@Aggregate(groupingExpressions, aExp, _) if validateAggregateExpressions(aExp) =>
         aExp.foreach {
           case alias: Alias =>
             validExpressionsMap ++= validateAggregateFunctionAndGetAlias(alias)
           case _: UnresolvedAlias =>
           case namedExpr: NamedExpression => validExpressionsMap.put(namedExpr.name, namedExpr)
         }
+        groupingExpressions foreach {
```

Comment:
I don't get the purpose of this change data loading rules is only to update the average column as average columns data will be stored in two column(sum, count), can u please explain why we need this change??

Reply:
This change is required for adding grouping expressions in logical plan for data loading in pre aggregate table as now expressions in group clause will also be treated as columns of pre aggregate table

Reply:
Ok, This is to handle when column is on grouping expression but not on projection, as we are adding column in data map user can create aggregation data map by adding that column in projection list.  But it is OK to support this type of expression in pre aggregate .

Reply:
As per CARBONDATA-1940, it was throwing an exception when grouping columns were not added in projection of preaggregate table